make build metadata opinionated

tiagralee opened the pull request:
issue #4 

- Add build metadata to the core and set default value in build time
- Removed the previous buildmetadata param in server initialise

andrewemeryanz commented on core/server.go:
```diff
@@ -5,13 +5,12 @@ package core
 )
 
 //nolint:gocognit // Long method names are okay because only generated code will call this, not humans.
-func Server(ctx context.Context, name string, hl Manager, grpcHl GrpcManager, logger *logrus.Logger, promRegistry *prometheus.Registry, buildMetadata *status.BuildMetadata) error {
```
This parameter can also be removed from `core.configureAdminServerListener`.

andrewemeryanz reviewed (changes requested):
One minor change, otherwise fine.

cuminandpaprika commented on core/build_metadata.go:
```diff
@@ -0,0 +1,40 @@
+	// supplied by the build environment at link time. Values are specified
+	// by passing "-X package.Varname=Value" to the linker. For example:
+	//
+	//  go build -ldflags="-X main.Name=BFF -X main.Version=1.0" ...etc...
```
Could we replace this example with something more generic?
e.g BFF isn't very clear as an application name. Something like `TestApp` or `DemoApp` could be more appropriate?

andrewemeryanz commented:
Looks good. Can you squash these changes into a single commit:
```sh
git reset --soft HEAD~3
git commit
```

and put a more descriptive commit message, something like:

```
Add build metadata

This change adds the ability to set build metadata at build time. Build
metadata is regarded as data that is common to all server builds (name, 
version, etc.).
```

cuminandpaprika reviewed (commented):
Looks good! 

One more thing that would be super helpful for any users is to document the need to run go build with the -ldflags, probably somewhere on the main sysl repo. If you don't want to jump on that right now, could you create an issue for it?

andrewemeryanz committed:
Add build metadata

This change adds the ability to set build metadata at build time. Build
metadata is regarded as data that is common to all server builds (name,
version, etc.).

Closes: #4

andrewemeryanz reviewed (changes requested):
This pull request now contains the changes from your other branch. I'll contact you on Slack to get this over the line.

andrewemeryanz commented:
> One more thing that would be super helpful for any users is to document ...

Issue created: https://github.com/anz-bank/sysl-go/issues/54

andrewemeryanz commented:
#major change.
